services/src/main/java/org/apache/druid/cli/CliOverlord.java

```diff
@@ -464,6 +455,38 @@ private void configureOverlordHelpers(Binder binder)
     );
   }
 
+  private void bindTaskLogStreamer(Binder binder)
+  {
+    String logsType = properties.getProperty("druid.indexer.logs.type");
+    if ("switching".equals(logsType)) {
+      bindSwitchingTaskLogs(binder);
```

Comment:
I am not entirely sure if we should be doing this.
For the Overlord, `SwitchingTaskLogs` should still be treated as one of the multiple bound log streamers that the `Overlord` would look at when asked for the logs of a task.

Maybe I am missing some use cases. Could you please explain this change?

Reply:
This has been removed now.